state/tx_options.go

```diff
@@ -0,0 +1,168 @@
+package state
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/celestiaorg/celestia-app/pkg/user"
+	apptypes "github.com/celestiaorg/celestia-app/x/blob/types"
+)
+
+const (
+	DefaultPrice float64 = -1.0
+	// gasMultiplier is used to increase gas limit in case if tx has additional options.
+	gasMultiplier = 1.1
+)
+
+func NewTxOptions(attributes ...Attribute) *TxOptions {
+	options := &TxOptions{gasPrice: DefaultPrice}
+	for _, attr := range attributes {
+		attr(options)
+	}
+	return options
+}
+
+// TxOptions specifies additional options that will be applied to the Tx.
+// Implements `Option` interface.
+type TxOptions struct {
+	// gasPrice represents the amount to be paid per gas unit.
+	// Negative gasPrice means user want us to use the minGasPrice
+	// defined in the node.
+	gasPrice float64
+	// since gasPrice can be 0, it is necessary to understand that user explicitly set it.
+	isGasPriceSet bool
+	// 0 gas means users want us to calculate it for them.
+	gas uint64
+
+	// Specifies the key from the keystore associated with an account that
+	// will be used to sign transactions.
+	// NOTE: This `Account` must be available in the `Keystore`.
+	accountKey string
```

Comment:
Sorry for chiming in so late on that but in my opinion it should (also) be possible to specificy the key by account address. Rationale: account addr is already a unique identifier which will always work no matter how the keys are stored or not. If people feel strongly about this, we can keep it as is.

Reply:
I agree with you. But from the user perspective imo it's easier to specify a key rather than deal with the address(imagine the user has 5-10 addresses). I'm fine with adding an alternative option `signerAddress` where users can specify the exact address. But let's agree on some cross cases:
in case the user provides both key + address and they do not match. We can:
1) return some misconfiguration error;
2) assume the address is a primary option;

cc @distractedm1nd